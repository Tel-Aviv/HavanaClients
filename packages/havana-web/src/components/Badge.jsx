import React from 'react';

const styles = {
    container: {
        position: 'relative',
        width: '100%',
        height: '100%'
    },
    badge: {
        borderRadius: '10px',
        display: 'inline-block',
        padding: '3px 7px',
        top: '-2px',
        right: '-2px',
        lineHeight: '1',
        color: '#fff',
        fontSize: '12px',
        fontWeight: '700',
        verticalAlign: 'baseline',
        textAlign: 'center',
        whiteSpace: 'nowrap',
        position: 'absolute',
        minWidth: '10px',
        backgroundColor: 'rgba(212,19,13,1)'
    }
}

const Badge = ({count}) => {
    return (
        <div>
            <div style={styles.container}>                                
                <span style={styles.badge}>
                    {count}
                </span>
            </div>
        </div>
    )
}

export default Badge;