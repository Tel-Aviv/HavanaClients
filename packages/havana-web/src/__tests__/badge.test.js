// 
// Pending Reports (Notifications) Test Suite
// 
// This test suite uses configuration approach to mock axios calls
// See setEnvVars.js file in conjuction with 
// setupFiles: ["<rootDir>/__tests__/setEnvVars.js"]
// in 'jest.config.js'
//
import 'babel-polyfill'

import React from 'react';
import renderer from 'react-test-renderer';
 
const API = require('../utils').API;
import Badge from '../components/Badge';

let count = 0;

describe('Get notifications count from API and display it on Badge component', () => {

    const OLD_ENV = process.env;

    beforeAll(async() => {

        //
        // Set 'mock' environment variable 
        // and load API dynamically that should be mocked
        //
        // jest.resetModules() // clears the cache
        // process.env = { ...OLD_ENV, 
        //                 mock: true }; // make a copy

        // Dynamically import API with 'mock' environemnt variable
        //const API = require('../utils').API;
        
        // Now call API
        const resp = await API.get('/me/pendings/count', {
            withCredentials: true
        })
        count = resp.data;
        expect(count).toBeGreaterThanOrEqual(0);
    })

    // afterAll(() => {
    //     process.env = OLD_ENV; // restore old env
    // });

    test.only('Test Badge Component', () => {
        const component = renderer.create(
            <Badge count={count} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})

