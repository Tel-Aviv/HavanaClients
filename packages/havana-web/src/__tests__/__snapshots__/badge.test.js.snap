// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get notifications count from API and display it on Badge component Test Badge Component 1`] = `
<div>
  <div
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <span
      style={
        Object {
          "backgroundColor": "rgba(212,19,13,1)",
          "borderRadius": "10px",
          "color": "#fff",
          "display": "inline-block",
          "fontSize": "12px",
          "fontWeight": "700",
          "lineHeight": "1",
          "minWidth": "10px",
          "padding": "3px 7px",
          "position": "absolute",
          "right": "-2px",
          "textAlign": "center",
          "top": "-2px",
          "verticalAlign": "baseline",
          "whiteSpace": "nowrap",
        }
      }
    >
      3
    </span>
  </div>
</div>
`;
